parser/match_operations_test.go

```diff
@@ -105,6 +105,79 @@ func TestMatchOperations(t *testing.T) {
 			},
 			err: false,
 		},
+		"simple transfer (with OppositeOrZeroAmounts)": {
+			operations: []*types.Operation{
+				{
+					Account: &types.AccountIdentifier{
+						Address: "addr2",
+					},
+					Amount: &types.Amount{
+						Value: "100",
+					},
+				},
+				{}, // extra op ignored
+				{
+					Account: &types.AccountIdentifier{
+						Address: "addr1",
+					},
+					Amount: &types.Amount{
+						Value: "-100",
+					},
+				},
+			},
+			descriptions: &Descriptions{
+				OppositeOrZeroAmounts: [][]int{{0, 1}},
+				OperationDescriptions: []*OperationDescription{
+					{
+						Account: &AccountDescription{
+							Exists: true,
+						},
+						Amount: &AmountDescription{
+							Exists: true,
+							Sign:   AnyAmountSign,
```

Comment:
Test passes when this is NegativeAmountSign and L146 is PositiveAmountSign - not sure why

Reply:
Actually - if I reverse the order of the Matches output expected for this test (i.e. the below), still using AnyAmountSign, it passes. This is also true of the other test I added (with 0 amount values). I'm unsure if changing the order changes the correctness of the result?

```
			matches: []*Match{
				{
					Operations: []*types.Operation{
						{
							Account: &types.AccountIdentifier{
								Address: "addr2",
							},
							Amount: &types.Amount{
								Value: "100",
							},
						},
					},
					Amounts: []*big.Int{big.NewInt(100)},
				},
				{
					Operations: []*types.Operation{
						{
							Account: &types.AccountIdentifier{
								Address: "addr1",
							},
							Amount: &types.Amount{
								Value: "-100",
							},
						},
					},
					Amounts: []*big.Int{big.NewInt(-100)},
				},
			},
			err: false,
		},
```